project/frontend/src/components/PreferenceForm.js

```diff
@@ -201,7 +191,9 @@ class PreferenceForm extends React.Component {
             onChange={this.changeState}
           />
         </label>
-        <button type='submit'>Submit</button>
+        <Link to={{ pathname: '/match-results', state: this.state }}>
```

Comment:
Is there a benefit to doing the form submit this way rather than the hadleSubmit function?

Reply:
We decided we want to send the form state to the new page and call `getRecommendations` in the `/match-results` page instead of within this form so that users don't press submit and then get stuck looking at their form for a second or two. So doing it this way, a user presses submit and goes straight to the `/match-results` page where a loading popup will render while we wait to get data from `getRecommendations`.

We could still do this using `handleSubmit` but instead of writing an extra event handler this `Link` component will do all the work for us. Also, using this `Link` makes sure we take advantage of React's one page application functionality wherein the user's browser wont actually refresh on submit and the page navigation is done "in-place" so there's no re-downloading content from the server.

Reply:
Does this still appropriately handle the required fields?

Reply:
Oh shoot it does not.

Reply:
Yeah I think the form onSubmit handles a lot of form related things under the hood. The event.preventDefault() in the handleSubmit will prevent the page from refreshing. Instead of executing getRecommendation in the handleSubmit, just redirect to the new page, with the values on state

Reply:
I'm thinking perhaps it doesn't matter. The only two required fields are latitude and longitude, and in the future these fields will be removed, and the user will choose their location by city/address (or let the browser detect their current location) before they can open the `PreferenceForm`. Once a user gets to the preference form I don't think we will be making any form fields required just in case the user wants to see any nearby restaurants (which would kind of defeat the purpose of the algorithm, but I don't think there's a point in taking away that functionality).

Reply:
I still think we should have a form onSubmit, just to avoid any undesirable side effects in the future. Having it this way with just a re-route link has the potential to contribute to bugs

Reply:
Ah okay got it, just seeing your previous message. Will refactor. Thanks!

Reply:
Adding onto what Jesse said, I would say it would be best to send the entire state object to the MatchResultsPage instead of parsing through each field in the state. Then you can just pass the whole object in as the preferences parameter in getRecommendation